Dockerfile

```diff
@@ -0,0 +1,50 @@
+FROM debian:latest AS builder
+
+ENV ZCASH_URL=https://github.com/zcash/zcash.git \
+    ZCASH_CONF=/home/zcash/.zcash/zcash.conf
+
+ARG VERSION=master
+
+RUN apt-get update
+
+RUN apt-get -qqy install --no-install-recommends build-essential \
+    automake ncurses-dev libcurl4-openssl-dev libssl-dev libgtest-dev \
+    make autoconf automake libtool git apt-utils pkg-config libc6-dev \
+    libcurl3-dev libudev-dev m4 g++-multilib unzip git python zlib1g-dev \
+    wget ca-certificates pwgen bsdmainutils curl
+
+WORKDIR /src
+RUN git clone ${ZCASH_URL}
+
+WORKDIR /src/zcash
+RUN git checkout $VERSION
+RUN ./zcutil/build.sh -j$(nproc)
+
+WORKDIR /src/zcash/src
+RUN /usr/bin/install -c zcash-tx zcashd zcash-cli zcash-gtest ../zcutil/fetch-params.sh -t /usr/bin/
+
+FROM debian:latest
+
+ENV ZCASH_CONF=/home/zcash/.zcash/zcash.conf
+
+RUN apt-get update
+
+RUN apt-get install -y libgomp1 wget curl
+
+RUN apt-get clean all -y
+
+COPY --from=builder /usr/bin/zcash-cli /usr/bin/zcashd /usr/bin/fetch-params.sh /usr/bin/
+
+RUN adduser --uid 1000 --system zcash && \
+    mkdir -p /home/zcash/.zcash/ && \
+    mkdir -p /home/zcash/.zcash-params/ && \
+    chown -R zcash /home/zcash && \
+    echo "Success"
+
+USER zcash
+RUN echo "rpcuser=zcash" > ${ZCASH_CONF} && \
+        echo "rpcpassword=`head /dev/urandom | tr -dc A-Za-z0-9 | head -c 13 ; echo ''`" >> ${ZCASH_CONF} && \
```

Comment:
How would this work if this built image was pushed to Dockerhub or built automatically? I see how this provides a nice default but wouldn't it be better to create this file outside of the build environment?

Reply:
The purpose behind doing this inside the build environment is that if one pulls the container, it will have all the minimum required defaults that zcashd needs to run without intervention.

In the documentation I will include the difference between using this default which is necessarily unsafe if one were to pull a pre-built container. That is why I overwrite the configuration file with the above `-v` with one that I keep outside the container.

If one just wanted to query the zcash blockchain they could use the default configuration file without mounting in any custom configurations.

Reply:
Yeh, that's what I assumed. My concern is simply that users won't change anything and use as a wallet but some explicit documentation in that regard should do the trick.

Reply:
Thinking about this some more an empty `zcash.conf` file has the same effect as it'll use cookie-based auth. Simply having the (empty) file present means it'll start a mainnet node.